app/merginapi.cpp

```diff
@@ -212,6 +232,19 @@ void MerginApi::deleteObsoleteFiles(QString projectPath)
     }
 }
 
+QString MerginApi::username() const
+{
+    return mUsername;
+}
+
+void MerginApi::setUsername(const QString &value)
+{
+    mUsername = value;
+    if (mUsername.isEmpty()) {
+        mToken.clear();
+    }
+}
```

Comment:
Do we need to keep a separate mUsername in addition to mToken? I would suggest to either: 1. keep just token and decode username from it when needed -or- 2. keep username+password and generate token from them when needed. The second option probably makes more sense...